geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionSingleHopDUnitTest.java

```diff
@@ -620,61 +666,63 @@ public void testMetadataFetchOnlyThroughputAll() {
     int redundantCopies = 3;
     int totalNumberOfBuckets = 4;
 
-    int port0 = vm0.invoke(() -> createServer(redundantCopies, totalNumberOfBuckets));
-    int port1 = vm1.invoke(() -> createServer(redundantCopies, totalNumberOfBuckets));
-    int port2 = vm2.invoke(() -> createServer(redundantCopies, totalNumberOfBuckets));
-    int port3 = vm3.invoke(() -> createServer(redundantCopies, totalNumberOfBuckets));
+    int port0 = vm0.invoke(() -> createServer(-1, redundantCopies, totalNumberOfBuckets));
+    int port1 = vm1.invoke(() -> createServer(-1, redundantCopies, totalNumberOfBuckets));
+    int port2 = vm2.invoke(() -> createServer(-1, redundantCopies, totalNumberOfBuckets));
+    int port3 = vm3.invoke(() -> createServer(-1, redundantCopies, totalNumberOfBuckets));
+    createClient(100, true, false, true, port0, port1, port2, port3);
 
-    createClient(port0, port1, port2, port3);
+    Region<Object, Object> partitionedRegion = getRegion(PARTITIONED_REGION_NAME);
 
-    putAll();
+    doPutAlls(partitionedRegion);
 
-    ClientMetadataService clientMetadataService = cache.getClientMetadataService();
+    ClientMetadataService clientMetadataService = CLIENT.get().getClientMetadataService();
     Map<String, ClientPartitionAdvisor> clientPRMetadata =
         clientMetadataService.getClientPRMetadata_TEST_ONLY();
+    ClientPartitionAdvisor prMetadata = clientPRMetadata.get(partitionedRegion.getFullPath());
 
     await().untilAsserted(() -> {
-      assertThat(clientPRMetadata).hasSize(1);
-    });
-
-    assertThat(clientPRMetadata).containsKey(testRegion.getFullPath());
-
-    ClientPartitionAdvisor prMetadata = clientPRMetadata.get(testRegion.getFullPath());
-
-    await().untilAsserted(() -> {
-      assertThat(prMetadata.getBucketServerLocationsMap_TEST_ONLY()).hasSize(totalNumberOfBuckets);
+      assertThat(clientPRMetadata)
+          .hasSize(1)
+          .containsKey(partitionedRegion.getFullPath());
+      assertThat(prMetadata.getBucketServerLocationsMap_TEST_ONLY())
+          .hasSize(totalNumberOfBuckets);
     });
   }
 
   @Test
   public void testMetadataIsSameOnAllServersAndClients() {
+    int locatorPort = DUnitEnv.get().getLocatorPort();
```

Comment:
dunitrunner revealed that testMetadataIsSameOnAllServersAndClients fails intermittently with underlying BindExceptions thrown when restarting server in vm0 or vm1 with the same port those two vms used to have a CacheServer running on.

The only way to fix this is to use a Locator in the pool instead of hardcoded server ports, and allow the CacheServers to come back up on any port.